test/tools/Modules/HttpListener/HttpListener.psm1

```diff
@@ -0,0 +1,154 @@
+            $listener.AuthenticationSchemes = [System.Net.AuthenticationSchemes]::Anonymous 
+            try {
+                Write-Warning "Note that thread is blocked waiting for a request.  After using Ctrl-C to stop listening, you need to send a valid HTTP request to stop the listener cleanly."
+                Write-Warning "Sending 'exit' command will cause listener to stop"
+                Write-Verbose "Listening on $url..."
+                $listener.Start()
+                $exit = $false
+                while ($exit -eq $false) {
+                    $context = $listener.GetContext()
+                    $request = $context.Request
+                    
+                    $queryItems = ParseQueryString($request.RawUrl)
+
+                    $test = $queryItems["test"]
+                    Write-Verbose "Testing: $test"
+
+                    # the status code to return for the response
+                    $statusCode = [System.Net.HttpStatusCode]::OK
+                    # this is the body of the response, return json/xml as appropriate
+                    $output = ""
+                    # this is hashtable of headers in the response
+                    $outputHeader = @{}
+
+                    switch ($test)
+                    {
+                        $null
+                        {
+                            $statusCode = [System.Net.HttpStatusCode]::BadRequest
+                            $output = "Test not specified"
+                        }
+                        "exit"
+                        {
+                            Write-Verbose "Received command to exit listener"
+                            $output = "Exit command received"
+                            $exit = $true
+                        }
+                        default
+                        {
+                            $statusCode = [System.Net.HttpStatusCode]::NotFound
+                            $output = "Unknown Test: $Test"
+                        }
+                    }
+
+                    $response = $context.Response
+                    $response.StatusCode = $statusCode
+                    $response.Headers.Clear()
+                    foreach ($header in $outputHeader.Keys)
+                    {
+                        $response.Headers.Add($header, $outputHeader[$header])
+                    }
+                    $buffer = [System.Text.Encoding]::UTF8.GetBytes($output)
+
+                    $response.ContentLength64 = $buffer.Length
+                    $output = $response.OutputStream
+                    $output.Write($buffer,0,$buffer.Length)
+                    $output.Close()
+                }
+            }
+            finally
+            {
+                $listener.Stop()
+            }
+        }
+
+        if ($AsJob)
+        {
+            Start-Job -ScriptBlock $script -ArgumentList $Port
+        }
+        else
+        {
+            & $script -Port $Port
+        }
+    }
+}
```

Comment:
from an automation perspective, I think this will be used more asynchronously, so I would rather see `AsJob` be the default. Perhaps `Blocking` should be the switch and default creates an asynchronous execution. Also, I'm also thinking that it might be better to run in a PowerShell instance rather than a job.

I think that the name of functions are too generic. If this is for our tests, then it should have that flavor, and be named `Start-TestHttpListener` or something along those lines